packages/expo-media-library/ios/EXMediaLibrary/EXMediaLibrary.m

```diff
@@ -924,7 +924,57 @@ + (NSString *)_assetUriForLocalId:(nonnull NSString *)localId
 + (PHAssetMediaType)_assetTypeForUri:(nonnull NSString *)localUri
 {
   CFStringRef fileExtension = (__bridge CFStringRef)[localUri pathExtension];
-  CFStringRef fileUTI = UTTypeCreatePreferredIdentifierForTag(kUTTagClassFilenameExtension, fileExtension, NULL);
+  CFStringRef fileUTI;
+  if (@available(iOS 14, *)) {
+    fileUTI = (__bridge CFStringRef)[[UTType typeWithFilenameExtension:(__bridge NSString*)fileExtension] identifier];
+  } else {
+    fileUTI = UTTypeCreatePreferredIdentifierForTag(kUTTagClassFilenameExtension, fileExtension, NULL);
+  }
+
+  // A fix for promise rejection on M1 simulator when run in Rosetta (as is by default as we don't compile ARM64 yet)
+  // https://developer.apple.com/forums/thread/702933
+  if(fileUTI == nil){
```

Comment:
```suggestion
  if (fileUTI == nil){
```

Also, could you move this logic to a separate method? Something like
`+ (PHAssetMediaType)assetTypeForFileExtension:(nonnull NSString *)fileExtension` should be fine (you can omit `_` for private methods, it's a relic from the past in our codebase)